daemon/docker.go

```diff
@@ -90,6 +92,23 @@ func (s *daemon) createDockerVolume(request *pluginRequest) (*Volume, error) {
 
 func (s *daemon) getDockerVolume(r *http.Request) (*Volume, *pluginRequest, error) {
 	request, err := convertToPluginRequest(r)
+	request.Opts = make(map[string]string)
+	//log.Debugf("Request obj is %v.\n Name:%s\n Opts:%v", request, request.Name, request.Opts)
+
+	//This check parses the name to check if there is a need to pick the size from the name.
+	sizere := regexp.MustCompile(".+@[1-9]+[0-9]*G$")
+	if sizere.MatchString(request.Name) {
+		log.Debugf("Volume name received (%s) needs to be parsed for size", request.Name)
+		vs := strings.Split(request.Name, "@")
```

Comment:
Use https://golang.org/pkg/strings/#LastIndex for the @ symbol and then there is no need to split and rejoin